File: VB/ViewModels/DocumentViewModelFactory.vb
Imports DXRibbonNotepad.Model

Namespace DXRibbonNotepad.ViewModels

    Public Module DocumentViewModelFactory

        Public Function CreateDefaultViewModel() As DocumentsViewModel
            Dim documents As Documents = CreateDefaultDocuments()
            Return CreateViewModel(documents)
        End Function

        Public Function CreateViewModel(ByVal model As Documents) As DocumentsViewModel
            Return New DocumentsViewModel(model)
        End Function
    End Module
End Namespace

File: CS/ViewModels/DocumentViewModelFactory.cs
using DXRibbonNotepad.Model;

namespace DXRibbonNotepad.ViewModels {
    public static class DocumentViewModelFactory {
        public static DocumentsViewModel CreateDefaultViewModel() {
            Documents documents = DocumentsFactory.CreateDefaultDocuments();
            return CreateViewModel(documents);
        }
        public static DocumentsViewModel CreateViewModel(Documents model) {
            return new DocumentsViewModel(model);
        }
    }
}